### ISSUE TITLE
Error closing position on Binance SPOT

### ISSUE THREAD

**Issue by rapmd73** [ccxt/ccxt#15712]
The is a very simple program that does nothing more then read the balance of an asset and sell it, ie. closing out the position. Selling at a fixed amount works fine, only when I try to use the balance, do I get the errors. I suspect the error is just a simple missing parameter, but a search of example code and documentation, as well as previous issues seems not to address this situation.

Thank you.

**=====[ Test code ]=====**

```#!/usr/bin/env python3
# -*- coding: utf-8 -*-

# Python3 vers. 3.8.10
# CCXT vers. 2.1.87

import sys
import ccxt

API="api"
SECRET="secret"

exchange=ccxt.binance()
exchange.apiKey=API
exchange.secret=SECRET
exchange.setSandboxMode(True)

exchange.load_markets()
exchange.verbose=True

pair='TRX/USDT'
base='TRX'

params={}

balance=exchange.fetch_balance()
bal=float(balance['total'][base])

print(bal)

# Fails with -1013 LOT_SIZE error
try:
    order=exchange.create_order(symbol=pair,type="market",side="sell",amount=bal)
    print(order)
except Exception as err:
    print(err)

# Fails with -1013 LOT_SIZE error
try:
    order=exchange.create_order(symbol=pair,type="market",side="sell",amount=int(bal))
    print(order)
except Exception as err:
    print(err)

# Fails with -2010 MARKET_LOT_SIZE error
try:
    params['quoteOrderQty']=bal
    order=exchange.create_order(symbol=pair,type="market",side="sell",amount=bal,params=params)
    print(order)
except Exception as err:
    print(err)

# Fails with -2010 MARKET_LOT_SIZE error
try:
    params['quoteOrderQty']=int(bal)
    order=exchange.create_order(symbol=pair,type="market",side="sell",amount=int(bal),params=params)
    print(order)
except Exception as err:
    print(err)
```

**===== [ Output ] =====**

fetch Request: binance GET https://testnet.binance.vision/api/v3/account?timestamp=1668614757689&recvWindow=10000&signature=e8a0249599b281d950ceccc855091af9d355b1c56193dc181ab2fa3d380eac69 RequestHeaders: {'X-MBX-APIKEY': 'ipD8kBZlisfLHfQVjw8EUO54Lux9fqy3wJzX8YVEUxyUYT5YkKts9KYt1YOijOPd', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: None

fetch Response: binance GET https://testnet.binance.vision/api/v3/account?timestamp=1668614757689&recvWindow=10000&signature=e8a0249599b281d950ceccc855091af9d355b1c56193dc181ab2fa3d380eac69 200 ResponseHeaders: {'Content-Type': 'application/json;charset=UTF-8', 'Content-Length': '271', 'Connection': 'keep-alive', 'Date': 'Wed, 16 Nov 2022 16:05:58 GMT', 'Server': 'nginx', 'x-mbx-uuid': '0aee5274-30fc-405d-a68a-a2fe57334ef0', 'x-mbx-used-weight': '44', 'x-mbx-used-weight-1m': '44', 'content-encoding': 'gzip', 'Strict-Transport-Security': 'max-age=31536000; includeSubdomains', 'X-Frame-Options': 'SAMEORIGIN', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Content-Security-Policy': "default-src 'self'", 'X-Content-Security-Policy': "default-src 'self'", 'X-WebKit-CSP': "default-src 'self'", 'Cache-Control': 'no-cache, no-store, must-revalidate', 'Pragma': 'no-cache', 'Expires': '0', 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Methods': 'GET, HEAD, OPTIONS', 'X-Cache': 'Miss from cloudfront', 'Via': '1.1 eb2d3c7af7f453242c6551f237e2edd4.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'MCI50-P1', 'X-Amz-Cf-Id': 'FHBB9Uv5itedwtI3e7eZsYJBBbXD2cX_iccuO5sMQz-XC8nB75IGgA=='} ResponseBody: {"makerCommission":0,"takerCommission":0,"buyerCommission":0,"sellerCommission":0,"canTrade":true,"canWithdraw":false,"canDeposit":false,"brokered":false,"updateTime":1668557066635,"accountType":"SPOT","balances":[{"asset":"BNB","free":"1000.00000000","locked":"0.00000000"},{"asset":"BTC","free":"1.00000000","locked":"0.00000000"},{"asset":"BUSD","free":"10000.00000000","locked":"0.00000000"},{"asset":"ETH","free":"100.00000000","locked":"0.00000000"},{"asset":"LTC","free":"500.00000000","locked":"0.00000000"},{"asset":"TRX","free":"484219.10000000","locked":"0.00000000"},{"asset":"USDT","free":"10789.20311300","locked":"0.00000000"},{"asset":"XRP","free":"50000.00000000","locked":"0.00000000"}],"permissions":["SPOT"]}
484219.1

fetch Request: binance POST https://testnet.binance.vision/api/v3/order RequestHeaders: {'X-MBX-APIKEY': 'ipD8kBZlisfLHfQVjw8EUO54Lux9fqy3wJzX8YVEUxyUYT5YkKts9KYt1YOijOPd', 'Content-Type': 'application/x-www-form-urlencoded', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: timestamp=1668614758141&symbol=TRXUSDT&side=SELL&type=MARKET&newClientOrderId=x-R4BD3S82e9de80e0e75f44c8fdf6ea&newOrderRespType=FULL&quantity=484219.1&recvWindow=10000&signature=4236fba9534aa22d615e212603786a53b6e459a0e88958beb1a5f430e51ae9c2

fetch Response: binance POST https://testnet.binance.vision/api/v3/order 400 ResponseHeaders: {'Content-Type': 'application/json;charset=UTF-8', 'Content-Length': '47', 'Connection': 'keep-alive', 'Date': 'Wed, 16 Nov 2022 16:05:58 GMT', 'Server': 'nginx', 'x-mbx-uuid': '51df79e9-d749-439b-8a92-3337f4ded623', 'x-mbx-used-weight': '45', 'x-mbx-used-weight-1m': '45', 'Strict-Transport-Security': 'max-age=31536000; includeSubdomains', 'X-Frame-Options': 'SAMEORIGIN', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Content-Security-Policy': "default-src 'self'", 'X-Content-Security-Policy': "default-src 'self'", 'X-WebKit-CSP': "default-src 'self'", 'Cache-Control': 'no-cache, no-store, must-revalidate', 'Pragma': 'no-cache', 'Expires': '0', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 eb2d3c7af7f453242c6551f237e2edd4.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'MCI50-P1', 'X-Amz-Cf-Id': '0jkidFNDH1CneE7jENpFVO4PBC1-ng-hDKpdGHkeTOynS8NhYcdrOw=='} ResponseBody: {"code":-1013,"msg":"Filter failure: LOT_SIZE"}
binance order amount should be evenly divisible by lot size {"code":-1013,"msg":"Filter failure: LOT_SIZE"}

fetch Request: binance POST https://testnet.binance.vision/api/v3/order RequestHeaders: {'X-MBX-APIKEY': 'ipD8kBZlisfLHfQVjw8EUO54Lux9fqy3wJzX8YVEUxyUYT5YkKts9KYt1YOijOPd', 'Content-Type': 'application/x-www-form-urlencoded', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: timestamp=1668614758592&symbol=TRXUSDT&side=SELL&type=MARKET&newClientOrderId=x-R4BD3S823aab6e9fb1c70669ba1a63&newOrderRespType=FULL&quantity=484219&recvWindow=10000&signature=7fe6d587274de813348a8513c9101fb04b8fbdcf7c2f1206142a9434f7919347

fetch Response: binance POST https://testnet.binance.vision/api/v3/order 400 ResponseHeaders: {'Content-Type': 'application/json;charset=UTF-8', 'Content-Length': '47', 'Connection': 'keep-alive', 'Date': 'Wed, 16 Nov 2022 16:05:58 GMT', 'Server': 'nginx', 'x-mbx-uuid': '51df79e9-d749-439b-8a92-3337f4ded623', 'x-mbx-used-weight': '46', 'x-mbx-used-weight-1m': '46', 'Strict-Transport-Security': 'max-age=31536000; includeSubdomains', 'X-Frame-Options': 'SAMEORIGIN', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Content-Security-Policy': "default-src 'self'", 'X-Content-Security-Policy': "default-src 'self'", 'X-WebKit-CSP': "default-src 'self'", 'Cache-Control': 'no-cache, no-store, must-revalidate', 'Pragma': 'no-cache', 'Expires': '0', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 eb2d3c7af7f453242c6551f237e2edd4.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'MCI50-P1', 'X-Amz-Cf-Id': '57pf-BnFy-U54yq49zgRkecm-f1RG4_nwR5IjGttpyDHjRXMo10IOA=='} ResponseBody: {"code":-1013,"msg":"Filter failure: LOT_SIZE"}
binance order amount should be evenly divisible by lot size {"code":-1013,"msg":"Filter failure: LOT_SIZE"}

fetch Request: binance POST https://testnet.binance.vision/api/v3/order RequestHeaders: {'X-MBX-APIKEY': 'ipD8kBZlisfLHfQVjw8EUO54Lux9fqy3wJzX8YVEUxyUYT5YkKts9KYt1YOijOPd', 'Content-Type': 'application/x-www-form-urlencoded', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: timestamp=1668614759070&symbol=TRXUSDT&side=SELL&type=MARKET&newClientOrderId=x-R4BD3S824919eed158ba0c6c9f36c4&newOrderRespType=FULL&quoteOrderQty=484219.1&recvWindow=10000&signature=4669be4ce420fb998d0dd57aa6fcb5cdbeb2776baf57d23a0cbf521d7b0fce75

fetch Response: binance POST https://testnet.binance.vision/api/v3/order 400 ResponseHeaders: {'Content-Type': 'application/json;charset=UTF-8', 'Content-Length': '54', 'Connection': 'keep-alive', 'Date': 'Wed, 16 Nov 2022 16:05:59 GMT', 'Server': 'nginx', 'x-mbx-uuid': '51df79e9-d749-439b-8a92-3337f4ded623', 'x-mbx-used-weight': '47', 'x-mbx-used-weight-1m': '47', 'x-mbx-order-count-10s': '1', 'x-mbx-order-count-1d': '13', 'Strict-Transport-Security': 'max-age=31536000; includeSubdomains', 'X-Frame-Options': 'SAMEORIGIN', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Content-Security-Policy': "default-src 'self'", 'X-Content-Security-Policy': "default-src 'self'", 'X-WebKit-CSP': "default-src 'self'", 'Cache-Control': 'no-cache, no-store, must-revalidate', 'Pragma': 'no-cache', 'Expires': '0', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 eb2d3c7af7f453242c6551f237e2edd4.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'MCI50-P1', 'X-Amz-Cf-Id': '9MS22mdWjYPpzJ06OQmAWweEaqdAxmkyqcle7NjF8N5J04y8SuirqA=='} ResponseBody: {"code":-2010,"msg":"Filter failure: MARKET_LOT_SIZE"}
binance order amount should be evenly divisible by lot size {"code":-2010,"msg":"Filter failure: MARKET_LOT_SIZE"}

fetch Request: binance POST https://testnet.binance.vision/api/v3/order RequestHeaders: {'X-MBX-APIKEY': 'ipD8kBZlisfLHfQVjw8EUO54Lux9fqy3wJzX8YVEUxyUYT5YkKts9KYt1YOijOPd', 'Content-Type': 'application/x-www-form-urlencoded', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: timestamp=1668614759684&symbol=TRXUSDT&side=SELL&type=MARKET&newClientOrderId=x-R4BD3S82ae3999783809533785ccf2&newOrderRespType=FULL&quoteOrderQty=484219&recvWindow=10000&signature=bfd88e8a0f43bed3118df9e4221aad6dbfac5a61def966234c613bf837c44072

fetch Response: binance POST https://testnet.binance.vision/api/v3/order 400 ResponseHeaders: {'Content-Type': 'application/json;charset=UTF-8', 'Content-Length': '54', 'Connection': 'keep-alive', 'Date': 'Wed, 16 Nov 2022 16:06:00 GMT', 'Server': 'nginx', 'x-mbx-uuid': '0aee5274-30fc-405d-a68a-a2fe57334ef0', 'x-mbx-used-weight': '1', 'x-mbx-used-weight-1m': '1', 'x-mbx-order-count-10s': '1', 'x-mbx-order-count-1d': '14', 'Strict-Transport-Security': 'max-age=31536000; includeSubdomains', 'X-Frame-Options': 'SAMEORIGIN', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Content-Security-Policy': "default-src 'self'", 'X-Content-Security-Policy': "default-src 'self'", 'X-WebKit-CSP': "default-src 'self'", 'Cache-Control': 'no-cache, no-store, must-revalidate', 'Pragma': 'no-cache', 'Expires': '0', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 eb2d3c7af7f453242c6551f237e2edd4.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'MCI50-P1', 'X-Amz-Cf-Id': 'u1pAr8KUQ2rpOfXy-1JIOqQ2mVz8ZsZBWT6oX3O9d2gMZ2BW8D4j6w=='} ResponseBody: {"code":-2010,"msg":"Filter failure: MARKET_LOT_SIZE"}
binance order amount should be evenly divisible by lot size {"code":-2010,"msg":"Filter failure: MARKET_LOT_SIZE"}

**Comment by Dan-krm** [comment]
> bal=float(balance['total'][base])

I'd recommend trying `bal=float(balance['free'][base])`

**Comment by rapmd73** [comment]
> > bal=float(balance['total'][base])
> 
> I'd recommend trying `bal=float(balance['free'][base])`

No effect to the error messages received. Thank you though for the idea is that would be more appropriate in a working environment.

**Comment by Dan-krm** [comment]
@rapmd73 is this for selling your spot holding, or for closing a spot-margin position?

**Comment by Dan-krm** [comment]
> try:
>     params['quoteOrderQty']=bal
>     order=exchange.create_order(symbol=pair,type="market",side="sell",amount=bal,params=params)
>     print(order)
> except Exception as err:
>     print(err)

Can you try `params['quantity']=bal` here instead of `params['quoteOrderQty']=bal` I think the quoteOrderQuantity might be expecting the value in USDT

https://binance-docs.github.io/apidocs/spot/en/#new-order-trade

**Comment by rapmd73** [comment]
> @rapmd73 is this for selling your spot holding, or for closing a spot-margin position?

Spot holding only.

**Comment by Dan-krm** [comment]
For this trading pair unfortunately Binance won't sell TRX past one decimal place so if you have '245.6785' available it can only sell a max of '245.6' you need to use `exchange.amount_to_precision(symbol, bal)` when sending a calculated amount value with multiple decimal places, try this:
```
def main():
    exchange = ccxt.binance({
        'apiKey': 'YOUR_API_KEY',
        'secret': 'YOUR_SECRET',
        # 'verbose': True,  # for debug output
    })
    exchange.load_markets()
    symbol = 'TRX/USDT'
    base = 'TRX'
    params = {}
    balance = exchange.fetch_balance()
    bal = float(balance['free'][base])
    convertedAmount = exchange.amount_to_precision(symbol, bal)

    try:
        order = exchange.create_order(symbol, 'market', 'sell', convertedAmount, params=params)
        print(order)
    except Exception as err:
        print(err)


if __name__ == '__main__':
    main()
```

**Comment by rapmd73** [comment]
I tried the changes you gave above. Here are the results including the modified test program. Thank you.

**===== [ Modified test program ] =====**

```
# -*- coding: utf-8 -*-

# Python3 vers. 3.8.10
# CCXT vers. 2.1.87

import sys
import ccxt

API="blah"
SECRET="blah"

exchange=ccxt.binance()
exchange.apiKey=API
exchange.secret=SECRET
exchange.setSandboxMode(True)

exchange.load_markets()
exchange.verbose=True

pair='TRX/USDT'
base='TRX'

params={}

balance=exchange.fetch_balance()
#bal=float(balance['total'][base])
bal=float(balance['free'][base])

bal=exchange.amount_to_precision(pair,bal)
print(bal)

# Fails with -1013 LOT_SIZE error
try:
    order=exchange.create_order(symbol=pair,type="market",side="sell",amount=bal)
    print(order)
except Exception as err:
    print(err)

# Fails with -2010 MARKET_LOT_SIZE error
try:
    params['quoteOrderQty']=bal
    order=exchange.create_order(symbol=pair,type="market",side="sell",amount=bal,params=params)
    print(order)
except Exception as err:
    print(err)

try:
    params['quantity']=bal
    order=exchange.create_order(symbol=pair,type="market",side="sell",amount=bal,params=params)
    print(order)
except Exception as err:
    print(err)
```

**===== [ Output ] =====**

fetch Request: binance GET https://testnet.binance.vision/api/v3/account?timestamp=1668643812728&recvWindow=10000&signature=91ccfed7491a72f2f610248350676e6358d18fc2fe864d4c96a2cbfd4f338cce RequestHeaders: {'X-MBX-APIKEY': 'ipD8kBZlisfLHfQVjw8EUO54Lux9fqy3wJzX8YVEUxyUYT5YkKts9KYt1YOijOPd', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: None

fetch Response: binance GET https://testnet.binance.vision/api/v3/account?timestamp=1668643812728&recvWindow=10000&signature=91ccfed7491a72f2f610248350676e6358d18fc2fe864d4c96a2cbfd4f338cce 200 ResponseHeaders: {'Content-Type': 'application/json;charset=UTF-8', 'Content-Length': '271', 'Connection': 'keep-alive', 'Date': 'Thu, 17 Nov 2022 00:10:13 GMT', 'Server': 'nginx', 'x-mbx-uuid': '51df79e9-d749-439b-8a92-3337f4ded623', 'x-mbx-used-weight': '20', 'x-mbx-used-weight-1m': '20', 'content-encoding': 'gzip', 'Strict-Transport-Security': 'max-age=31536000; includeSubdomains', 'X-Frame-Options': 'SAMEORIGIN', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Content-Security-Policy': "default-src 'self'", 'X-Content-Security-Policy': "default-src 'self'", 'X-WebKit-CSP': "default-src 'self'", 'Cache-Control': 'no-cache, no-store, must-revalidate', 'Pragma': 'no-cache', 'Expires': '0', 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Methods': 'GET, HEAD, OPTIONS', 'X-Cache': 'Miss from cloudfront', 'Via': '1.1 b0a64e9cc5e3b7be7927374083991434.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'MCI50-P1', 'X-Amz-Cf-Id': 'yEOINTtLzqHFcsexJrdY8Y_pIhBGa5icmXFdsygJ_QM73Jk6ik-ltw=='} ResponseBody: {"makerCommission":0,"takerCommission":0,"buyerCommission":0,"sellerCommission":0,"canTrade":true,"canWithdraw":false,"canDeposit":false,"brokered":false,"updateTime":1668643071583,"accountType":"SPOT","balances":[{"asset":"BNB","free":"1000.00000000","locked":"0.00000000"},{"asset":"BTC","free":"1.00000000","locked":"0.00000000"},{"asset":"BUSD","free":"10000.00000000","locked":"0.00000000"},{"asset":"ETH","free":"100.00000000","locked":"0.00000000"},{"asset":"LTC","free":"500.00000000","locked":"0.00000000"},{"asset":"TRX","free":"492151.70000000","locked":"0.00000000"},{"asset":"USDT","free":"10390.90726300","locked":"0.00000000"},{"asset":"XRP","free":"50000.00000000","locked":"0.00000000"}],"permissions":["SPOT"]}
492151.7

fetch Request: binance POST https://testnet.binance.vision/api/v3/order RequestHeaders: {'X-MBX-APIKEY': 'ipD8kBZlisfLHfQVjw8EUO54Lux9fqy3wJzX8YVEUxyUYT5YkKts9KYt1YOijOPd', 'Content-Type': 'application/x-www-form-urlencoded', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: timestamp=1668643813189&symbol=TRXUSDT&side=SELL&type=MARKET&newClientOrderId=x-R4BD3S82a981184835d803d70aab9d&newOrderRespType=FULL&quantity=492151.7&recvWindow=10000&signature=7f3aaab15f1c644eb953ed163fd17e010bf62976bac15a809d954eae40407778

fetch Response: binance POST https://testnet.binance.vision/api/v3/order 400 ResponseHeaders: {'Content-Type': 'application/json;charset=UTF-8', 'Content-Length': '47', 'Connection': 'keep-alive', 'Date': 'Thu, 17 Nov 2022 00:10:13 GMT', 'Server': 'nginx', 'x-mbx-uuid': '0aee5274-30fc-405d-a68a-a2fe57334ef0', 'x-mbx-used-weight': '21', 'x-mbx-used-weight-1m': '21', 'Strict-Transport-Security': 'max-age=31536000; includeSubdomains', 'X-Frame-Options': 'SAMEORIGIN', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Content-Security-Policy': "default-src 'self'", 'X-Content-Security-Policy': "default-src 'self'", 'X-WebKit-CSP': "default-src 'self'", 'Cache-Control': 'no-cache, no-store, must-revalidate', 'Pragma': 'no-cache', 'Expires': '0', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 b0a64e9cc5e3b7be7927374083991434.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'MCI50-P1', 'X-Amz-Cf-Id': '2zhf4H2h2ZfmRhO8myDY01QL6j5_2bgxfUDYAm7AWvA2I_4KHDNxqg=='} ResponseBody: {"code":-1013,"msg":"Filter failure: LOT_SIZE"}
binance order amount should be evenly divisible by lot size {"code":-1013,"msg":"Filter failure: LOT_SIZE"}

fetch Request: binance POST https://testnet.binance.vision/api/v3/order RequestHeaders: {'X-MBX-APIKEY': 'ipD8kBZlisfLHfQVjw8EUO54Lux9fqy3wJzX8YVEUxyUYT5YkKts9KYt1YOijOPd', 'Content-Type': 'application/x-www-form-urlencoded', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: timestamp=1668643813645&symbol=TRXUSDT&side=SELL&type=MARKET&newClientOrderId=x-R4BD3S82cf502cd484e3d39c312ed3&newOrderRespType=FULL&quoteOrderQty=492151.7&recvWindow=10000&signature=5aafdb7429a8fc4af397c6c90bf93bae398a7d6cf78b017a01effdf9cde8bcb3

fetch Response: binance POST https://testnet.binance.vision/api/v3/order 400 ResponseHeaders: {'Content-Type': 'application/json;charset=UTF-8', 'Content-Length': '54', 'Connection': 'keep-alive', 'Date': 'Thu, 17 Nov 2022 00:10:14 GMT', 'Server': 'nginx', 'x-mbx-uuid': '0aee5274-30fc-405d-a68a-a2fe57334ef0', 'x-mbx-used-weight': '22', 'x-mbx-used-weight-1m': '22', 'x-mbx-order-count-10s': '1', 'x-mbx-order-count-1d': '13', 'Strict-Transport-Security': 'max-age=31536000; includeSubdomains', 'X-Frame-Options': 'SAMEORIGIN', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Content-Security-Policy': "default-src 'self'", 'X-Content-Security-Policy': "default-src 'self'", 'X-WebKit-CSP': "default-src 'self'", 'Cache-Control': 'no-cache, no-store, must-revalidate', 'Pragma': 'no-cache', 'Expires': '0', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 b0a64e9cc5e3b7be7927374083991434.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'MCI50-P1', 'X-Amz-Cf-Id': 'sBje13ZyOjkPtdFn20rIroB23upTyYGKfmhC7vFmllyl0jWEir8jUQ=='} ResponseBody: {"code":-2010,"msg":"Filter failure: MARKET_LOT_SIZE"}
binance order amount should be evenly divisible by lot size {"code":-2010,"msg":"Filter failure: MARKET_LOT_SIZE"}

fetch Request: binance POST https://testnet.binance.vision/api/v3/order RequestHeaders: {'X-MBX-APIKEY': 'ipD8kBZlisfLHfQVjw8EUO54Lux9fqy3wJzX8YVEUxyUYT5YkKts9KYt1YOijOPd', 'Content-Type': 'application/x-www-form-urlencoded', 'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: timestamp=1668643814102&symbol=TRXUSDT&side=SELL&type=MARKET&newClientOrderId=x-R4BD3S82310cecc13418b029a0471&newOrderRespType=FULL&quoteOrderQty=492151.7&quantity=492151.7&recvWindow=10000&signature=f82cdb3544c58bbc08e52c020cca9de7ce5fb189127413ce34c1f88098a96577

fetch Response: binance POST https://testnet.binance.vision/api/v3/order 400 ResponseHeaders: {'Content-Type': 'application/json;charset=UTF-8', 'Content-Length': '72', 'Connection': 'keep-alive', 'Date': 'Thu, 17 Nov 2022 00:10:14 GMT', 'Server': 'nginx', 'x-mbx-uuid': '0aee5274-30fc-405d-a68a-a2fe57334ef0', 'x-mbx-used-weight': '23', 'x-mbx-used-weight-1m': '23', 'Strict-Transport-Security': 'max-age=31536000; includeSubdomains', 'X-Frame-Options': 'SAMEORIGIN', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Content-Security-Policy': "default-src 'self'", 'X-Content-Security-Policy': "default-src 'self'", 'X-WebKit-CSP': "default-src 'self'", 'Cache-Control': 'no-cache, no-store, must-revalidate', 'Pragma': 'no-cache', 'Expires': '0', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 b0a64e9cc5e3b7be7927374083991434.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'MCI50-P1', 'X-Amz-Cf-Id': 'J8cwF9WwH1yxVUOYA9JsHRlqogKMCcWzBNIRwEDOag-bQtH0M8K6RA=='} ResponseBody: {"code":-1106,"msg":"Parameter 'quoteOrderQty' sent when not required."}
binance {"code":-1106,"msg":"Parameter 'quoteOrderQty' sent when not required."}

**Comment by rapmd73** [comment]
I just figured out and discovered through various sizes of trying to sell that Binance rate limits how much you can sell based upon your KYC status.

The test net only allows $500 at a time to be sold. Anything above that causes a market lot size error.

I've looked through the messages and so far I can't seem to find anything that indicates a maximum position size based upon KYC status.

**Comment by Dan-krm** [comment]
I couldn't find any specific details on trading limits based on KYC tiers but it's possible that this page is updated and user specific: https://www.binance.com/en/trade-rule

You can search the limits and filters that are different based on if it's the testnet or live trading:
```
    exchange = ccxt.binance({
        'apiKey': 'YOUR_API_KEY',
        'secret': 'YOUR_SECRET',
    })
    exchange.set_sandbox_mode(True)
    exchange.load_markets()
    symbol = 'TRX/USDT'
    market = exchange.market(symbol)
    filters = market['info']['filters']
    print(market)
    print(filters)
```

https://binance-docs.github.io/apidocs/spot/en/#filters
https://github.com/ccxt/ccxt/wiki/Manual#precision-and-limits
https://github.com/ccxt/ccxt/wiki/Manual#market-structure

Closing this issue now, feel free to reopen the issue if you have any more related questions

**Comment by rapmd73** [comment]
I just finished around of tests with the test net and determined that the limit for non-kyc is $505. Anything above that for selling amount causes a market size lot error.

The test net provides a generous amount of various assets so it made it very easy to determine this by using a random number generator between $499 and 510. Everything above 505 caused errors while everything from 499 to 505 succeeded.

I tried to find specific requirements with the KYC process as well and only came across the document you mentioned. I could find nothing within the filters it actually said with the limit is on the basis of the KYC, so I suspect this is just going to be trial and error based upon each individual using the test net to find out what their actual limits really are.

Thank you.
